src/spikeinterface/postprocessing/template_similarity.py

```diff
@@ -147,10 +154,144 @@ def _get_data(self):
 compute_template_similarity = ComputeTemplateSimilarity.function_factory()
 
 
+if HAVE_NUMBA:
+
+    @numba.jit(nopython=True, parallel=True, fastmath=True, nogil=True)
+    def _compute_similarity_matrix(templates_array, other_templates_array, num_shifts, mask, method):
+        num_templates = templates_array.shape[0]
+        num_samples = templates_array.shape[1]
+        other_num_templates = other_templates_array.shape[0]
+
+        num_shifts_both_sides = 2 * num_shifts + 1
+        distances = np.ones((num_shifts_both_sides, num_templates, other_num_templates), dtype=np.float32)
+        same_array = np.array_equal(templates_array, other_templates_array)
+
+        # We can use the fact that dist[i,j] at lag t is equal to dist[j,i] at time -t
+        # So the matrix can be computed only for negative lags and be transposed
+
+        if same_array:
+            # optimisation when array are the same because of symetry in shift
+            shift_loop = range(-num_shifts, 1)
+        else:
+            shift_loop = range(-num_shifts, num_shifts + 1)
+
+        for count, shift in enumerate(shift_loop):
+            src_sliced_templates = templates_array[:, num_shifts : num_samples - num_shifts]
+            tgt_sliced_templates = other_templates_array[:, num_shifts + shift : num_samples - num_shifts + shift]
+            for i in numba.prange(num_templates):
+                src_template = src_sliced_templates[i]
+                overlapping_templates = np.flatnonzero(np.sum(mask[i], 1))
+                tgt_templates = tgt_sliced_templates[overlapping_templates]
+                for gcount, j in enumerate(overlapping_templates):
+
+                    # symmetric values are handled later
+                    if same_array and j < i:
+                        # no need exhaustive looping when same template
+                        continue
+                    src = src_template[:, mask[i, j]].flatten()
+                    tgt = (tgt_templates[gcount][:, mask[i, j]]).flatten()
+
+                    norm_i = 0
+                    norm_j = 0
+
+                    for k in range(len(src)):
+                        if method == "l1":
```

Comment:
Maybe we should use constant with number to avoid testing against a str object in any inner loops.
 Something in C style like
```python
L1 = 0
L2 = 1
COSINE = 2

method_int = {"l1": L1, "l2": L2, "cosine": COSINE}[method]

for ... in ... :
    for ... in ... :
       for ... in ... :
            if method_int == L1:




```

Reply:
I've made something like that, but no speed difference